### DISCUSSION TITLE
Suggestion Needed

### DISCUSSION THREAD

**Discussion by sharukhkhan234** [horilla-opensource/horilla#273] [Q&A]
Greetings,

Need suggestions about when to update the software with the latest code base, I have connected Horila with Postgres and working on it on a regular basis, and there is some data in it. therefore I want to update the code, 

should I execute this command " pip install -r requirements.txt" and rest commands or re-clone it,

**Comment by horilla-opensource** [comment]
Hi @sharukhkhan234 ,
Basically you should start by either cloning or pulling the repo first to a directory. Once that's done, you'll get the latest code in the repo at that time into your local repo.
Whenever a new commits are added, if you want the change to be applied to your local repo, execute the command `git pull origin master` and run the `makemigrations` and `migrate` command.
In case there is any changes to the `requirements.txt` file, run the `pip3 install -r requirements.txt` commmand.

**Reply by horilla-opensource** [reply]
Make sure that you don't remove the files from the `migrations` folder inside each app as it might have some issues to the database if removed.

**Reply by sharukhkhan234** [reply]
should I run this command "git pull origin master" in the "enviroment/horilla" directory or in the environment?

**Reply by horilla-opensource** [reply]
In the directory where git is initialized. ie the directory where all the apps are present.

**Reply by sharukhkhan234** [reply]
by running this command, does it affect the existing settings or data?

**Reply by horilla-opensource** [reply]
No this doesn't the existing data.

**Reply by sharukhkhan234** [reply]
Thank You for the kind help!

need one more help regarding employees, As Admin can see, create, delete, and update all the employees, However, I want that department shouldn't have permission to see all the employees on the employee's dashboard. I mean to say that only the "HR" department and they are the admin can have all kinds of info, The rest departments shouldn't have permission to see all the employees on the dashboard, and I think there's no permission category to exclude that one from the dashboard.

Regards,
